### PR TITLE
Clarify in-app strings

### PR DESCRIPTION
**Pull request by KobeW50**

### CODE REVIEW

File: src/main/resources/music/settings/host/values/strings.xml [hunk 1]
```diff
@@ -119,8 +119,8 @@ Info:
     <string name="revanced_replace_flyout_menu_dismiss_queue_watch_on_youtube_warning">Invalid video url.</string>
     <string name="revanced_replace_flyout_menu_dismiss_queue_title">Replace dismiss queue</string>
     <string name="revanced_replace_flyout_menu_dismiss_queue_summary">Replaces \'Dismiss queue\' with \'Watch on YouTube\'.</string>
-    <string name="revanced_replace_flyout_menu_report_only_player_title">Applies only to player flyout menu</string>
-    <string name="revanced_replace_flyout_menu_report_only_player_summary">Report menu in comments will not be replaced.</string>
+    <string name="revanced_replace_flyout_menu_report_only_player_title">Keep report in comments</string>
+    <string name="revanced_replace_flyout_menu_report_only_player_summary">Replaces \'Report\' only in the player flyout menu.</string>
```
Comment (KobeW50) [inline src/main/resources/music/settings/host/values/strings.xml]: Two issues with this change: 

1. It isn't immediately clear how the description of the setting is relevant to the title.

2. "Replaces" somewhat implies that there would be a Playback speed menu in the comments section, which ofc isn't true.

File: src/main/resources/music/settings/host/values/strings.xml [hunk 2]
```diff
@@ -243,8 +243,9 @@ This does not bypass the age restriction. It just accepts it automatically."</st
     <string name="revanced_restore_old_player_layout_summary">"Returns the player layout to the old style.
 Some features may not work properly in the old player layout."</string>
     <string name="revanced_replace_player_cast_button_title">Replace cast button</string>
-    <string name="revanced_replace_player_cast_button_summary">"Replace the cast button in the player with the 'Open music' button. (Experimental)
-Limitation: If one or more music that cannot be played in the playlist, it does not work normally."</string>
+    <string name="revanced_replace_player_cast_button_summary">"Replaces the cast button in the player with the 'Open music' button. (Experimental)
+
+Limitation: It does not work normally if any songs in the playlist cannot be played."</string>
```
Comment (KobeW50) [inline src/main/resources/music/settings/host/values/strings.xml]: I don't know what the 'Open music' button does since I haven't gotten it to work. 

Instead of a "Limitation" section, it can have an "Info" section explaining what the 'Open music' button does and this limitation. This would be very similar to the Trim silence setting, shown below.

image


Additionally, in the limitation it says "it does not work normally..." but it is unclear what "it" is referring to. ("It" could mean this setting, or it could mean the button.)

File: src/main/resources/music/settings/host/values/strings.xml [hunk 3]
```diff
@@ -357,7 +358,7 @@ Limitation: If one or more music that cannot be played in the playlist, it does
     <string name="revanced_enable_debug_logging_title">Enable debug logging</string>
     <string name="revanced_enable_debug_logging_summary">Prints the debug log.</string>
     <string name="revanced_enable_debug_buffer_logging_title">Enable debug buffer logging</string>
-    <string name="revanced_enable_debug_buffer_logging_summary">Prints the debug logs include buffer.</string>
+    <string name="revanced_enable_debug_buffer_logging_summary">Includes the buffer in the debug log.</string>
```
Comment (KobeW50) [inline src/main/resources/music/settings/host/values/strings.xml:361]: IDk if this new sentence makes sense. Please advise

File: src/main/resources/music/settings/host/values/strings.xml (hunk 2 above)
Comment (inotia00) [reply src/main/resources/music/settings/host/values/strings.xml]: Explanation of this setting can be found in https://github.com/inotia00/ReVanced_Extended/issues/1431.
I have given up on fixing the issues associated with this feature, so this feature will be removed in the near future.

File: src/main/resources/music/settings/host/values/strings.xml [hunk 4]
```diff
@@ -243,8 +243,9 @@ This does not bypass the age restriction. It just accepts it automatically."</st
     <string name="revanced_restore_old_player_layout_summary">"Returns the player layout to the old style.
 Some features may not work properly in the old player layout."</string>
     <string name="revanced_replace_player_cast_button_title">Replace cast button</string>
-    <string name="revanced_replace_player_cast_button_summary">"Replace the cast button in the player with the 'Open music' button. (Experimental)
-Limitation: If one or more music that cannot be played in the playlist, it does not work normally."</string>
+    <string name="revanced_replace_player_cast_button_summary">"Replaces the cast button in the player with the 'Open music' button. (Experimental)
+
+Info: This feature does not work normally if any songs in the playlist cannot be played."</string>
```
Comment (KobeW50) [inline src/main/resources/music/settings/host/values/strings.xml]: How do you feel about one of these?

```suggestion
    <string name="revanced_replace_player_cast_button_summary">"Replaces the cast button in the player with an 'Open music' button, which opens the official audio instead of the music video. (Experimental)

Info: This feature does not work normally if any songs in the playlist cannot be played."</string>
```

Or alternatively:

```suggestion
    <string name="revanced_replace_player_cast_button_summary">"Replaces the cast button in the player with an 'Open music' button. (Experimental)

Info: 
• The 'Open music' button opens the official audio instead of the music video audio.
• This feature does not work normally if any songs in the playlist cannot be played."</string>
```

**Comment by KobeW50** [comment]
In YT, under ReVanced Extended > Feed > Video filter, the Hide recommended videos setting description says it hides videos with the 'Only for membership' tag. 

Screenshot_20240512_170502_YT Extended

On my client I have a 'Members only' tag, shown below.

Screenshot_20240512_170356_YouTube

Are these the same, and should the setting description say `Videos with the 'Members only' tag.`?

**Comment by KobeW50** [comment]
More questions: 

**1.** The spoof format stream data setting says it has the following limitation:

"There may be about 5 seconds of buffering when the video starts."

Would this be better? "There may be about 5 seconds of buffering **when videos start**." 

___

**2.** In the Alternative thumbnails settings, there is an option for "Player playlists, recommendations"

What exactly does this mean? (Example below)

Screenshot_20240512_161259_YT Extended


___

**3.**

In ReVanced Extended > Player > Player buttons, there is the experimental setting of `Override download action button`.

This setting is somewhat out-of-place because it is for the download button below the video rather than the overlay download button.

The current description is shown in this screenshot:



Should I change it to the following?  

"Download button **below the video player** opens the native in-app downloader."

Or do you intend to move it to the Action button settings? (I can open a bug report)

Or neither?

___

**4.**

In the Player settings, the note for the `Disable speed overlay` setting says that, "This setting does not force the speed overlay to be enabled."

What does this mean? (Image below for context)



___

**5.**

What does this setting do?



___

**6.**

What does this setting do?

File: src/main/resources/youtube/settings/host/values/strings.xml [hunk 5]
```diff
@@ -347,16 +347,16 @@ Some components may not be hidden."</string>
...
 
     <!-- PreferenceScreen: General, PreferenceCategory: General, PreferenceScreen: Navigation buttons -->
     <string name="revanced_preference_screen_navigation_buttons_title">Navigation buttons</string>
     <string name="revanced_preference_screen_navigation_buttons_summary">Hide or show navigation bar section components.</string>
 
     <string name="revanced_enable_narrow_navigation_buttons_title">Enable narrow navigation buttons</string>
-    <string name="revanced_enable_narrow_navigation_buttons_summary_on">Spacing between navigation buttons becomes narrower.</string>
-    <string name="revanced_enable_narrow_navigation_buttons_summary_off">Spacing between navigation buttons does not become narrower.</string>
+    <string name="revanced_enable_narrow_navigation_buttons_summary_on">Spacing between navigation buttons is narrow.</string>
+    <string name="revanced_enable_narrow_navigation_buttons_summary_off">Spacing between navigation buttons is not narrow.</string>
```
Comment (KobeW50) [inline src/main/resources/youtube/settings/host/values/strings.xml]: Alternatively:

```suggestion
    <string name="revanced_enable_narrow_navigation_buttons_summary_off">Spacing between navigation buttons is wide.</string>
```

Or 

```suggestion
    <string name="revanced_enable_narrow_navigation_buttons_summary_off">Spacing between navigation buttons is normal.</string>
```

File: src/main/resources/youtube/settings/host/values/strings.xml [hunk 6]
```diff
@@ -379,31 +379,31 @@ Some components may not be hidden."</string>
     <string name="revanced_hide_navigation_label_summary_on">Navigation label is hidden.</string>
     <string name="revanced_hide_navigation_label_summary_off">Navigation label is shown.</string>
     <string name="revanced_switch_create_with_notifications_button_title">Switch create with notifications</string>
-    <string name="revanced_switch_create_with_notifications_button_summary">"Switch the positions of the create button and notification button by spoofing device information.
+    <string name="revanced_switch_create_with_notifications_button_summary">"Swaps the positions of the create button and notification button by spoofing device information.
 
-• Even if you change this setting, it may not take effect until you reboot the device.
+• If this setting is changed, the device may need to be rebooted for it to take effect.
```
Comment (KobeW50) [inline src/main/resources/youtube/settings/host/values/strings.xml]: Alternatively: 
```suggestion
• The device may need to be rebooted for a change of this setting to take effect.
```

File: src/main/resources/youtube/settings/host/values/strings.xml [hunk 7]
```diff
@@ -379,31 +379,31 @@ Some components may not be hidden."</string>
...
 
     <!-- PreferenceScreen: General, PreferenceCategory: General, PreferenceScreen: Toolbar -->
     <string name="revanced_preference_screen_toolbar_title">Toolbar</string>
-    <string name="revanced_preference_screen_toolbar_summary">Hide or change components located on the toolbar such as toolbar buttons, search bar, header.</string>
+    <string name="revanced_preference_screen_toolbar_summary">Hide or change components located on the toolbar, such as the search bar, toolbar buttons, and header.</string>
 
     <string name="revanced_change_youtube_header_title">Change YouTube header</string>
     <string name="revanced_change_youtube_header_summary_on">Premium header is enabled.</string>
-    <string name="revanced_change_youtube_header_summary_off">Generic headers is enabled.</string>
+    <string name="revanced_change_youtube_header_summary_off">Generic header is enabled.</string>
```
Comment (KobeW50) [inline src/main/resources/youtube/settings/host/values/strings.xml:406]: I wasn't sure if the pluralization is deliberate (since iirc YT occasionally has themed headers). 

If it should be kept plural, it should be as follows:

```suggestion
    <string name="revanced_change_youtube_header_summary_off">Generic headers are enabled.</string>
```

File: src/main/resources/youtube/settings/host/values/strings.xml [hunk 8]
```diff
@@ -1072,20 +1072,21 @@ Limitation: Official headers in search results will be hidden."</string>
     <string name="revanced_enable_default_playback_speed_shorts_summary_off">Default playback speed does not apply to Shorts.</string>
     <string name="revanced_skipped_preloaded_buffer">Skipped preloaded buffer.</string>
     <string name="revanced_skip_preloaded_buffer_title">Skip preloaded buffer</string>
-    <string name="revanced_skip_preloaded_buffer_summary">"Skip preloaded buffer at video start to bypass default video quality enforcement delay.
+    <string name="revanced_skip_preloaded_buffer_summary">"Skips the preloaded buffer at the start of videos to immediately apply the default video quality.
```
Comment (KobeW50) [inline src/main/resources/youtube/settings/host/values/strings.xml:1074]: Double-check that this explanation is accurate.

**Comment by inotia00** [comment]
> In YT, under ReVanced Extended > Feed > Video filter, the Hide recommended videos setting description says it hides videos with the 'Only for membership' tag.

If I remember correctly, this is the feature suggested by @YT-Advanced.
He may know more accurately.

**Comment by inotia00** [comment]
> **1.** The spoof format stream data setting says it has the following limitation:
> 
> Would this be better? "There may be about 5 seconds of buffering when videos start."

: Yes this explanation is clear

> **2.** In the Alternative thumbnails settings, there is an option for "Player playlists, recommendations"
> 
> What exactly does this mean? (Example below)

: This setting was added when merging [ReVanced's commit](https://github.com/ReVanced/revanced-patches/blob/3a229763d02ffa40a367c9337e2a373d2d7cda3e/src/main/resources/addresources/values/strings.xml#L970)
I think it's a thumbnail included in `Playlists` and [Horizontal shelves](https://github.com/ReVanced/revanced-patches/blob/3a229763d02ffa40a367c9337e2a373d2d7cda3e/src/main/resources/addresources/values/strings.xml#L101) that appear in the feed.


> **3.** In ReVanced Extended > Player > Player buttons, there is the experimental setting of `Override download action button`.
> Should I change it to the following?
> "Download button below the video player opens the native in-app downloader."
> Or do you intend to move it to the Action button settings? (I can open a bug report)
> Or neither?

: This setting overrides the behavior of both the download button in the action bar and the download button in the feed flyout menu. (Unlike YouTube Music)

Therefore, it is more appropriate to separate it into the `External downloader` category.

> **4.** In the Player settings, the note for the Disable speed overlay setting says that, "This setting does not force the speed overlay to be enabled."
> 
> What does this mean? (Image below for context)

: The code related to the speed overlay was added in YouTube v18.19.36+, but Google restricted it to be activated only in YouTube v18.36.39+. (server side limitation)

Therefore, for users who have patched YouTube v18.29.38 ~ v18.35.36, the speed overlay will not be activated even if you enable or disable the `Disable speed overlay` setting.

Likewise, when spoofing to old clients with the same settings as Spoof app version, the speed overlay is not activated.

One additional line of explanation has been added to prevent user confusion.

> **5.** What does this setting do?

: `Expand video description` setting is the most unstable and buggy of the `Experimental Flags`.

To prevent operation in unintended situations, a process has been added to check the panel's title string. (refer to screenshot)

Description

The panel's title string depends on the user's system language, and the video description will only expand if the correct panel's title string is entered.

> **6.** What does this setting do?

: If I were to explain this, I would have to explain how the patch is implemented.
(It's a little complicated.)

When the value of the `Keep landscape mode` flag is TRUE, the screen orientation can only be changed by the 'fullscreen button'.
(The screen orientation cannot be changed in any way other than that.)

If the flag value is always TRUE, the user cannot switch to landscape mode (fullscreen) using back button or swipe gesture.

Therefore, the value of the flag always remains FALSE until the user enters fullscreen, turns off the screen, and turns the screen back on.

After the screen is turned on, the flag value switches to TRUE, and after the screen is unlocked, the flag value must be restored to FALSE.
(To prevent issues where the screen orientation does not change through the back button or swipe gesture)

The problem is that there is no way to check whether the user has unlocked the screen.

Therefore, I adopted a method of changing the value of the flag to FALSE after a set time.

In other words, this is the time interval from 'When the user turns on the screen' to 'When the value of the flag is restored to FALSE'.

If this time interval is too short, fullscreen may not remain when the user unlocks it.

If this time interval is too long, the user will not be able to change the screen orientation via the back button or swipe gesture after unlocking.

**Comment by KobeW50** [comment]
> 3 
> : This setting overrides the behavior of both the download button in the action bar and the download button in the feed flyout menu. (Unlike YouTube Music)
> 
> Therefore, it is more appropriate to separate it into the `External downloader` category.

Is it ok if I change the setting descriptions to "**Native** download button opens the native in-app downloader." and "**Native** download button opens your external downloader."?

**Comment by inotia00** [comment]
> "**Native** download button opens your external downloader."

This looks better

**Comment by KobeW50** [comment]
> > **2.** In the Alternative thumbnails settings, there is an option for "Player playlists, recommendations"
> 
> : This setting was added when merging [ReVanced's commit](https://github.com/ReVanced/revanced-patches/blob/3a229763d02ffa40a367c9337e2a373d2d7cda3e/src/main/resources/addresources/values/strings.xml#L970) I think it's a thumbnail included in `Playlists` and [Horizontal shelves](https://github.com/ReVanced/revanced-patches/blob/3a229763d02ffa40a367c9337e2a373d2d7cda3e/src/main/resources/addresources/values/strings.xml#L101) that appear in the feed.

After a short amount of testing, this option changes the thumbnails of the end-card recommendations, but I am not certain if this is all that it does. I asked the author of the [upstream PR](https://github.com/ReVanced/revanced-patches/pull/2926/files#r1599361741) and am awaiting a response. Once its effects are confirmed, I will try to make the setting title clearer (in a separate PR).

Other than that, this PR is ready for review.

**Comment by inotia00** [comment]
Thanks
